### PR TITLE
Create borrowing endpoint

### PR DESCRIPTION
**Pull request by achalaidiuk**

### CODE REVIEW

File: borrowings/serializers.py [hunk 1]
```diff
@@ -20,3 +24,34 @@ class BorrowingCreateSerializer(serializers.ModelSerializer):
     class Meta:
         model = Borrowing
         fields = ["id", "book", "expected_return_date", "actual_return_date"]
+
+    def validate(self, attrs):
```
Comment (MasakDirt) [inline borrowings/serializers.py:28]: add type annotations

File: borrowings/serializers.py [hunk 2]
```diff
@@ -20,3 +24,34 @@ class BorrowingCreateSerializer(serializers.ModelSerializer):
...
+                and expected_return_date
+                and expected_return_date < borrow_date
+        ):
+            raise serializers.ValidationError(
+                "Expected return date cannot be earlier than borrow date."
+            )
+
+        return attrs
+
+    def create(self, validated_data):
```
Comment (MasakDirt) [inline borrowings/serializers.py:46]: type annotations

File: borrowings/tests/test_borrowing_api.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 from books.models import Book
 from borrowings.models import Borrowing
-from datetime import date, timedelta
+from datetime import date, timedelta, datetime
 
 BORROWINGS_URL = reverse("borrowings:borrowings-list")
```
Comment (MasakDirt) [inline borrowings/tests/test_borrowing_api.py:13]: add space between imports and constant

File: borrowings/serializers.py (hunk 1 above)
Comment (MasakDirt) [inline borrowings/serializers.py:28]: make this validate function in other file and add it to models

**Review by MasakDirt — CHANGES_REQUESTED** [review]

File: borrowings/models.py [hunk 4]
```diff
@@ -12,5 +14,27 @@ class Borrowing(models.Model):
     actual_return_date = models.DateField(null=True, blank=True)
 
     def __str__(self):
```
Comment (MasakDirt) [inline borrowings/models.py:16]: missing return type annotation

File: borrowings/serializers.py [hunk 5]
```diff
@@ -20,3 +28,28 @@ class BorrowingCreateSerializer(serializers.ModelSerializer):
     class Meta:
         model = Borrowing
         fields = ["id", "book", "expected_return_date", "actual_return_date"]
+
+    def validate(self, attrs) -> dict:
+        expected_return_date = attrs.get("expected_return_date")
+
+        if expected_return_date is not None:
```
Comment (MasakDirt) [inline borrowings/serializers.py:35]: Is expected return date can be non here?

File: borrowings/serializers.py [hunk 6]
```diff
@@ -20,3 +28,28 @@ class BorrowingCreateSerializer(serializers.ModelSerializer):
...
+
+        return attrs
+
+    def create(self, validated_data) -> Borrowing:
+        with transaction.atomic():
+            book_data = validated_data.pop("book")
+            book = Book.objects.get(id=book_data.id)
+            if book.inventory < 1:
+                raise ValidationError("the following book is not available")
+            borrowing = Borrowing.objects.create(
```
Comment (MasakDirt) [inline borrowings/serializers.py:50]: You add `transaction.atomic()` too early, I think we can add it before this line on 49 line

File: borrowings/serializers.py [hunk 7]
```diff
@@ -20,3 +28,28 @@ class BorrowingCreateSerializer(serializers.ModelSerializer):
...
+                expected_return_date=expected_return_date,
+                error_to_raise=serializers.ValidationError
+            )
+
+        return attrs
+
+    def create(self, validated_data) -> Borrowing:
+        with transaction.atomic():
+            book_data = validated_data.pop("book")
+            book = Book.objects.get(id=book_data.id)
```
Comment (MasakDirt) [inline borrowings/serializers.py:47]: here is better use get_object_or_404

File: borrowings/tests/test_borrowing_api.py (hunk 3 above)
Comment (MasakDirt) [inline borrowings/tests/test_borrowing_api.py:14]: add space between imports and constant

File: borrowings/validation.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
+def validate_borrowing(
+        borrow_date,
```
Comment (MasakDirt) [inline borrowings/validation.py:2]: add type annotations

**Review by MasakDirt — CHANGES_REQUESTED** [review]

**Review by MasakDirt — APPROVED** [review]

**Review by AndriiHamasa — APPROVED** [review]

**Pull request merged**